src/strands/multiagent/swarm.py

```diff
@@ -0,0 +1,646 @@
+    final_result: str | None = None
+
+
+class Swarm(MultiAgentBase):
+    """Self-organizing collaborative agent teams with shared working memory."""
+
+    def __init__(
+        self,
+        nodes: list[Agent],
+        config: SwarmConfig | None = None,
+    ) -> None:
+        """Initialize Swarm with agents and configuration.
+
+        Args:
+            nodes: List of nodes (e.g. Agent) to include in the swarm
+            config: Optional swarm execution configuration
+        """
+        super().__init__()
+
+        self.config = config or SwarmConfig()
+        self.shared_context = SharedContext()
+        self.nodes: dict[str, SwarmNode] = {}
+        self.state = SwarmState(
+            current_node=SwarmNode("", Agent()),  # Placeholder, will be set properly
+            task="",
+            completion_status=Status.PENDING,
+        )
+
+        self._setup_swarm(nodes)
+        self._inject_swarm_tools()
+
+    def execute(self, task: str) -> SwarmResult:
+        """Execute task synchronously."""
+
+        def execute() -> SwarmResult:
+            return asyncio.run(self.execute_async(task))
+
+        with ThreadPoolExecutor() as executor:
+            future = executor.submit(execute)
+            return future.result()
+
+    async def execute_async(self, task: str) -> SwarmResult:
+        """Execute the swarm asynchronously."""
+        logger.info("starting swarm execution")
+
+        # Initialize swarm state with configuration
+        initial_node = next(iter(self.nodes.values()))  # First SwarmNode
+        self.state = SwarmState(
+            current_node=initial_node,
+            task=task,
+            completion_status=Status.EXECUTING,
+            shared_context=self.shared_context,
+        )
+        self.shared_context.set_task(task)
+
+        start_time = time.time()
+        try:
+            logger.info("current_node=<%s> | starting swarm execution with node", self.state.current_node.node_id)
+            logger.info(
+                "max_handoffs=<%d>, max_iterations=<%d>, timeout=<%s>s | SwarmConfig",
+                self.config.max_handoffs,
+                self.config.max_iterations,
+                self.config.execution_timeout,
+            )
+
+            await self._execute_swarm()
+
+            if self.state.completion_status == Status.EXECUTING:
+                self.state.completion_status = Status.COMPLETED
+
+        except Exception:
+            logger.exception("swarm execution failed")
+            self.state.completion_status = Status.FAILED
+            raise
+        finally:
+            self.state.execution_time = round((time.time() - start_time) * 1000)
+
+        return self._build_result()
+
+    def _setup_swarm(self, nodes: list[Agent]) -> None:
+        """Initialize swarm configuration."""
+        # Validate agents have names and create SwarmNode objects
+        for i, node in enumerate(nodes):
+            if not hasattr(node, "name") or not node.name:
```

Comment:
Here we should/can use `node.name` directly, no?  No need to be extra safe with `hasattr`?